src/plugin/memory-provider.ts

```diff
@@ -53,133 +44,55 @@ export class MemoryProvider {
             const contributedTracker = handlerForSession?.initializeAdapterTracker?.(session);
 
             return ({
-                onWillStartSession: () => {
-                    this.debugSessionStarted(session);
-                    contributedTracker?.onWillStartSession?.();
-                },
-                onWillStopSession: () => {
-                    this.debugSessionTerminated(session);
-                    contributedTracker?.onWillStopSession?.();
-                },
-                onDidSendMessage: message => {
-                    if (isDebugResponse('initialize', message)) {
-                        // Check for right capabilities in the adapter
-                        this.sessionDebugCapabilities.set(session.id, message.body);
-                        if (vscode.debug.activeDebugSession?.id === session.id) {
-                            this.setContext(session);
-                        }
-                    } else if (isDebugEvent('stopped', message)) {
-                        this._onDidStopDebug.fire(session);
-                    } else if (isDebugEvent('memory', message)) {
-                        delete this.scheduledOnDidMemoryWriteEvents[message.body.memoryReference];
-                        this._onDidWriteMemory.fire(message.body);
-                    }
-                    contributedTracker?.onDidSendMessage?.(message);
-                },
-                onError: error => { contributedTracker?.onError?.(error); },
-                onExit: (code, signal) => { contributedTracker?.onExit?.(code, signal); },
-                onWillReceiveMessage: message => {
-                    if (isDebugRequest('initialize', message)) {
-                        this.sessionClientCapabilities.set(session.id, message.arguments);
-                    }
-                    contributedTracker?.onWillReceiveMessage?.(message);
-                }
+                onWillStartSession: () => contributedTracker?.onWillStartSession?.(),
```

Comment:
I think that this is now equivalent to `return contributedTracker` since we don't actually do any work of our own here. It looks like it may be intended to facilitate #97, but if so, it might be better to put this code in a PR addressing that issue.

Reply:
I had to adapt the return type of the function but you are of course correct. I'd even say that it is more correct now as we simply return undefined if we do not have an adapter tracker instead of an empty adapter tracker that never does anything. I really like this.